src/main/java/net/glowstone/entity/GlowPlayer.java

```diff
@@ -2826,10 +2846,47 @@ public GlowBlock getDigging() {
         return digging;
     }
 
+    private void broadcastBlockBreakAnimation(GlowBlock block, int destroyStage) {
+        if (block == null) return;
+
+        GlowChunk.Key key = new GlowChunk.Key(block.getChunk().getX(), block.getChunk().getZ());
+        // TODO: should canSeeChunk filtering take place here, in GlowPlayer, or both?
+        block.getWorld().getRawPlayers().stream().filter(player1 -> player1.canSeeChunk(key)).forEach(player2 -> {
```

Comment:
It is getting the chunk of the block and filtering the list of players to those who can see the chunk, so it only sends the block breaking animation to the right players. I wasn't familiar with the Streams API and Java lambdas but it seems to be a common pattern in Glowstone:

```
src/main/java/net/glowstone/block/entity/BlockEntity.java
45:        block.getWorld().getRawPlayers().stream().filter(player -> player.canSeeChunk(key)).forEach(this::update);

src/main/java/net/glowstone/block/state/GlowNoteBlock.java
189:        getWorld().getRawPlayers().stream().filter(player -> player.canSeeChunk(key)).forEach(player -> player.playNote(location, instrument, note));

src/main/java/net/glowstone/entity/GlowEntity.java
1024:        world.getRawPlayers().stream().filter(player -> player.canSeeEntity(this)).forEach(player -> player.getSession().send(message));

src/main/java/net/glowstone/entity/objects/GlowItem.java
71:            world.getRawPlayers().stream().filter(other -> other.canSeeEntity(this)).forEach(other -> other.getSession().send(message));
```

but the players are filtered twice for chunk visibility, both here and in sendBlockBreakAnimation:

```java
    public void sendBlockBreakAnimation(BlockBreakAnimationMessage message) {
        // only send message if the chunk is within visible range
        Key key = new Key(message.getX() >> 4, message.getZ() >> 4);
        if (canSeeChunk(key)) {
            afterBlockChanges.add(message);
        }
    }
```

leaning towards removing this second check and only performing the filtering in broadcastBlockBreakAnimation